src/nvim/screen.c

```diff
@@ -7116,7 +7116,24 @@ void ui_ext_tabline_update(void)
 
     ADD(tabs, DICTIONARY_OBJ(tab_info));
   }
-  ui_call_tabline_update(curtab->handle, tabs);
+
+  Array buffers = ARRAY_DICT_INIT;
+  FOR_ALL_BUFFERS(buf) {
+    // Do not include unlisted buffers
+    if (!buf->b_p_bl) {
```

Comment:
```
$ git grep b_p_bl '*.h'
src/nvim/buffer_defs.h:682:  int b_p_bl;                   ///< 'buflisted'
```

Unset buflisted on some buffer in a new test.
Create the test in `test/functional/ui/tabline_spec.lua`.